File: templates/spring-api/src/main/java/com/felixmurcia/api/domain/valueobjects/Email.java
// templates/spring-api/src/main/java/com/felixmurcia/api/domain/valueobjects/Email.java
package com.felixmurcia.api.domain.valueobjects;
public record Email(String value) {
    public Email {
        var pattern = "^[\\w-.]+@([\\w-]+\\.)+[\\w-]{2,4}$";
        if (!value.matches(pattern)) throw new IllegalArgumentException("Invalid email format");
        value = value.toLowerCase().trim();
    }
}

File: templates/fastapi-api/src/domain/value_objects/email.py
# templates/fastapi-api/src/domain/value_objects/email.py
import re

class Email:
    def __init__(self, value: str):
        pattern = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
        if not re.match(pattern, value):
            raise ValueError(f"Invalid email format: {value}")
        self.value = value.lower().strip()

    def __eq__(self, other):
        return isinstance(other, Email) and self.value == other.value
    
    def __hash__(self):
        return hash(self.value)
